apps/polls/assets/PollManagement.jsx

```diff
@@ -0,0 +1,224 @@
+var api = require('adhocracy4').api
+var React = require('react')
+var django = require('django')
+var update = require('react-addons-update')
+var FlipMove = require('react-flip-move')
+var QuestionForm = require('./QuestionForm')
+
+let PollManagement = React.createClass({
+  getInitialState: function () {
+    return {
+      questions: this.props.poll.questions,
+      errors: [],
+      successMessage: '',
+      maxQuestionKey: 0,
+      maxChoiceKey: 0
+    }
+  },
+
+  /*
+  |--------------------------------------------------------------------------
+  | Question state related handlers
+  |--------------------------------------------------------------------------
+  */
+
+  getNextQuestionKey: function () {
+    /** Get an artifical key for non-commited questions.
+     *
+     *  Prefix to prevent collisions with real database keys;
+     */
+    var questionKey = 'local_' + (this.state.maxQuestionKey + 1)
+    this.setState({maxQuestionKey: this.state.maxQuestionKey + 1})
+    return questionKey
```

Comment:
good point, i changed it to be a class variable. the changes should be reflected in documents, too